.github/workflows/main.yml

```diff
@@ -88,9 +56,19 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           push: true
-          tags: ${{ steps.meta.outputs.tags }}
+          tags: "${{ env.IMAGE_NAME }}:${{ github.sha }}"
           labels: ${{ steps.meta.outputs.labels }}
 
+      - name: Run Trivy vulnerability scanner
```

Comment:
important: moving Trivy scanner to this part of the pipeline, will not prevent a vulnerable docker image to be uploaded into the registry. I strongly recommend you execute trivy on a locally build image before pushing it.